hyperspy/roi.py

```diff
@@ -83,17 +83,51 @@ def __init__(self):
             """, arguments=['roi'])
         self.signal_map = dict()
 
+    def __getitem__(self, *args, **kwargs):
+        return self.values.__getitem__(*args, **kwargs)
+
+    def __repr__(self):
+        para = []
+        for name, value in self.parameters.items():
+            if value is t.Undefined:
+                para.append(f"{name}={value}")
+            else:
+                # otherwise format value with the General specifer
+                para.append(f"{name}={value:G}")
+        return f"{self.__class__.__name__}({', '.join(para)})"
+
     _ndim = 0
     ndim = property(lambda s: s._ndim)
 
+    @property
+    def parameters(self):
+        raise NotImplementedError()
+
+    @property
+    def values(self):
```

Comment:
Since the same can be done with `tuple(roi)` or `list(roi)`, I don't think this property is necessary. Same reason as in my other comment.

Reply:
Yep.